python/ray/data/_internal/logical/operators/input_data_operator.py

```diff
@@ -26,24 +27,20 @@ def __init__(
         )
         self.input_data = input_data
         self.input_data_factory = input_data_factory
-        self._output_metadata_cache = None
 
     def output_data(self) -> Optional[List[RefBundle]]:
         if self.input_data is None:
             return None
         return self.input_data
 
     def aggregate_output_metadata(self) -> BlockMetadata:
-        """Get aggregated output metadata, using cache if available."""
+        return self._cached_output_metadata
+
+    @functools.cached_property
```

Comment:
Can we directly apply `@functools.cached_property` to `aggregate_output_metadata`?

Reply:
Yes, but I tried not to break the old interface. If we use `functools.cached_property` on the `aggregate_output_metadata` method, all callers would need to change from `obj.aggregate_output_metadata()` to `obj.aggregate_output_metadata`. Do you think that's okay?